MetaMask/metamask-extension
# dom tame poc

works nicely, still needs to tame addEventListener (EventMessage.prototype window leaking props).
test by:
* `yarn dist`
* lookup `document.createElement("div");` under the entire `dist/chrome` folder
* pick the one within `common-1.js` (should have the string "50px" right after it)
* paste the following code right after it
* visit the extension with your devtools open

```javascript
;(function test1(){ debugger;
    try {
        document.createElement('a').ownerDocument.location
    } catch (e) {
        e.message === 'NOT ALLOWED(document): location';
    }
    try {
        document.createElement('a').ownerDocument.open
    } catch (e) {
        e.message === 'NOT ALLOWED(document): open';
    }
    try {
        document.createElement('a').ownerDocument.defaultView.alert
    } catch (e) {
        e.message === 'NOT ALLOWED(window): alert';
    }
    try {
        document.createElement('a').ownerDocument.body.getRootNode() === document.createElement('a').ownerDocument
    } catch {}
    try {
        document.createElement('a').ownerDocument.body.parentNode.parentNode === document.createElement('a').ownerDocument
    } catch {}
}());
```
